programs/timefn.c

```diff
@@ -95,10 +92,19 @@ UTIL_time_t UTIL_getTime(void)
     /* time must be initialized, othersize it may fail msan test.
      * No good reason, likely a limitation of timespec_get() for some target */
     UTIL_time_t time = UTIL_TIME_INITIALIZER;
+#if defined(TIMEFN_HAS_TIMESPEC_GET)
+    /* favor timespec_get() as first choice for c11 targets */
```

Comment:
For testing: could we add `asm("\n# Using timespec_get")` and `asm("\n# Using clock_gettime")` to each branch of the ifdef? Then we could grep the generated assembly for those strings and ensure only one string is included.

(I'm not sure exactly how to insert a comment or symbol into the assembly, took the above code from https://stackoverflow.com/a/28493396/3154996. I think it is definitely possible, though).

Reply:
That would add some code specifically for this test.
It would have to be enabled through some dedicated setting (like a macro) since we don't want that for the general case.

Then indeed it would be possible to couple that generation with an external tool that grep the outcome and ensure it's consistent. This `asm("# message")` thing seems to only leave a comment in the generated assembly. So that means accessing the generated assembly and grep it.

That looks plausible.

Reply:
It seems like we can get text into the actual binary like this: `asm volatile(".global message \n message:")` ([godbolt](https://godbolt.org/z/4crhcYYK9)). I can test if this works for our situation and if so put up a PR which does it (gated to `DEBUGLEVEL >= 2` or something). Agree that getting the generated assembly would require some work, hopefully that's not necessary.

Reply:
Yes, this would be a welcome test capability.
As it stands, this PR is stuck as long as there is no way to reliably test `timefn` interface and ensure it's interpreted the same way in all units that include it.